project/src/main/java/com/google/sps/data/ServletLibrary.java

```diff
@@ -0,0 +1,165 @@
+package com.google.sps.data;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.vision.v1.*;
+
+import java.util.*;
+ 
+import java.io.IOException;
+
+public class ServletLibrary{
+
+
+
+    public static ArrayList<Productset> listProductSets(String projectId, String computeRegion) throws IOException {
+
+    ArrayList<Productset> productsets = new ArrayList<>();
+
+    try (ProductSearchClient client = ProductSearchClient.create()) {
+        // A resource that represents Google Cloud Platform location.
+        String formattedParent = ProductSearchClient.formatLocationName(projectId, computeRegion);
+        // List all the product sets available in the region.
+        for (ProductSet productSet : client.listProductSets(formattedParent).iterateAll()) {
+        // Display the product set information
+        System.out.println(String.format("Product set name: %s", productSet.getName()));
+        System.out.println(
+            String.format(
+                "Product set id: %s",
+                productSet.getName().substring(productSet.getName().lastIndexOf('/') + 1)));
+        System.out.println(
+            String.format("Product set display name: %s", productSet.getDisplayName()));
+        System.out.println("Product set index time:");
+        System.out.println(String.format("\tseconds: %s", productSet.getIndexTime().getSeconds()));
+        System.out.println(String.format("\tnanos: %s", productSet.getIndexTime().getNanos()));
+
+        Productset productset = new Productset(productSet.getName().substring(productSet.getName().lastIndexOf('/') + 1),
```

Comment:
A comment explaining what exactly you are doing here would be useful, especially with the lastIndexOf and the +1.